glide-core/src/client/value_conversion.rs

```diff
@@ -607,6 +608,64 @@ pub(crate) fn convert_to_expected_type(
             }
             _ => Err((ErrorKind::TypeError, "Response couldn't be converted").into()),
         },
+        // Used by XAUTOCLAIM. The command returns a list of length 2 if the Redis version is less than 7.0.0 or a list
+        // of length 3 otherwise. It has the following response format:
+        /* Redis version < 7.0.0 example:
+        1) "0-0"
+        2) 1) 1) "1-0"
+              2) 1) "field1"
+                 2) "value1"
+                 3) "field2"
+                 4) "value2"
+           2) 1) "1-1"
+              2) 1) "field3"
+                 2) "value3"
+           3) (nil)  // Entry IDs that were in the Pending Entry List but no longer in the stream get a nil value.
+           4) (nil)  // These nil values will be dropped so that we can return a map value for the second response element.
+
+        Redis version >= 7.0.0 example:
+        1) "0-0"
+        2) 1) 1) "1-0"
+              2) 1) "field1"
+                 2) "value1"
+                 3) "field2"
+                 4) "value2"
+           2) 1) "1-1"
+              2) 1) "field3"
+                 2) "value3"
+        3) 1) "1-2"  // Entry IDs that were in the Pending Entry List but no longer in the stream are listed in the
+           2) "1-3"  // third response element, which is an array of these IDs.
+        */
+        ExpectedReturnType::XAutoClaimReturnType => match value {
+            // Response will have 2 elements if Redis version < 7.0.0, and 3 elements otherwise.
+            Value::Array(mut array) if array.len() == 2 || array.len() == 3 => {
+                let mut result: Vec<Value> = Vec::with_capacity(array.len());
+                result.push(array[0].clone());
+
+                let mut stale_entry_ids: Option<Value> = None;
```

Comment:
instead of creating stale_entries_ids, why not do this below:
```
                if array.len() == 3 {
                         result.push(Some(array.remove(2));
                }
```

Reply:
I think its better this way for performance because I am always removing from the end of the array. The third element of the array is an array of strings. On line 651 I remove the second element, and if I haven't removed the third element yet, the remove call will have to shift the third element into the position at index 1. I'm wondering if that means the third element needs to be copied and thus it isn't as ideal for performance? Let me know your thoughts

Reply:
what about `array[1].clone()` and `array[2].clone()`?

Reply:
I think that would be worse performance wise since we would be copying both elements, I was asking about this above: https://github.com/aws/glide-for-redis/pull/1718#discussion_r1660044842